include/nbl/video/IGPUDescriptorSet.h

```diff
@@ -112,49 +118,39 @@ class NBL_API IGPUDescriptorSet : public asset::IDescriptorSet<const IGPUDescrip
             return samplers + localOffset;
         }
 
-        inline uint32_t getDescriptorStorageOffset(const asset::E_DESCRIPTOR_TYPE type) const { return m_descriptorStorageOffsets.data[type]; }
-        inline uint32_t getMutableSamplerStorageOffset() const { return m_descriptorStorageOffsets.data[asset::EDT_COUNT]; }
-
-	protected:
-		virtual ~IGPUDescriptorSet()
-		{
-            for (auto i = 0u; i < asset::EDT_COUNT; ++i)
-            {
-                // There is no descriptor of such type in the set.
-                if (m_descriptorStorageOffsets.data[i] == ~0u)
-                    continue;
+        inline core::smart_refctd_ptr<asset::IDescriptor>* getAllDescriptors(const asset::IDescriptor::E_TYPE type) const
+        {
+            auto* baseAddress = m_pool->getDescriptorStorage(type);
+            if (baseAddress == nullptr)
+                return nullptr;
 
-                const auto type = static_cast<asset::E_DESCRIPTOR_TYPE>(i);
-                std::destroy_n(getDescriptorStorage(type) + m_descriptorStorageOffsets.data[i], m_layout->getTotalDescriptorCount(type));
-            }
+            const auto offset = m_descriptorStorageOffsets.data[static_cast<uint32_t>(type)];
```

Comment:
seems like you should be using `getDescriptorStorageOffset` for self documenting code :)

Reply:
Resolved.